cmd/local_client.go

```diff
@@ -94,12 +94,12 @@ func updateConfig(config strpkg.Config, debug bool) strpkg.Config {
 }
 
 func logNodeBalance(store *strpkg.Store) {
-	balance, err := presenters.ShowEthBalance(store)
+	balance, kv, err := presenters.ShowEthBalance(store)
```

Comment:
Rather than returning a 3 tuple, could we only return the `kv` map, and place `message` inside of it as a third key? You would then just use the value when needed: `kv["message"]`.

The standard 2 tuple followed by an `err` is so idiomatic in golang that I'm reluctant to leave it. Open to other ideas.

Reply:
Before this, I triggered the log entry in the presenter method, rather than returning the message and key/values. Only issue was the presenter tests couldn't assert the message and key values returned.

The tests for user controller also asserts the balance amounts returned from the same method, since that's already covered, we could fire the log in the presenter and just assert that the method doesn't error in the test?

Reply:
As you said, we have tests that assert more than just the lack of an error, such as https://github.com/smartcontractkit/chainlink/pull/618/files#diff-cfc3843eaf893d1fa809d266f0199abdR96 which checks balances, which is great. I'm not sure which controller tests you're speaking of that also check balance. Furthermore, the `presenters` package isn't responsible for output, but rather molding data types for easier presentation, whether that be json log, GUI, or tabular output. In this example, that's then consumed by `cmd` `local_client` which is a better place to be writing output.

I think we should stick with a `kv` that has the three keys you need: `message`, `balance`, `address`.